src/bosh-director/lib/bosh/blobstore_client/gcscli_blobstore_client.rb

```diff
@@ -0,0 +1,125 @@
+require 'securerandom'
+require 'open3'
+require 'json'
+
+module Bosh::Blobstore
+  class GcscliBlobstoreClient < BaseClient
+
+    EXIT_CODE_OBJECT_NOTFOUND = 3
+
+    # Blobstore client for GCS, using bosh-gcscli binary
+    # @param [Hash] options GCSconnection options
+    # @option options [Symbol] bucket_name
+    #   key that is applied before the object is sent to GCS
+    # @option options [Symbol, optional] access_key_id
+    # @option options [Symbol, optional] secret_access_key
+    # @option options [Symbol] gcscli_path
+    #   path to gcscli binary
+    # @option options [Symbol, optional] gcscli_config_path
+    #   path to store configuration files
+    # @note If access_key_id and secret_access_key are not present, the
+    #   blobstore client operates in read only mode
+    def initialize(options)
+      super(options)
+
+      @gcscli_path = @options.fetch(:gcscli_path)
+      unless Kernel.system("#{@gcscli_path}", "--v", out: "/dev/null", err: "/dev/null")
+        raise BlobstoreError, "Cannot find gcscli executable. Please specify gcscli_path parameter"
+      end
+
+      @gcscli_options = {
+        bucket_name: @options[:bucket_name],
+        credentials_source: @options.fetch(:credentials_source, 'none'),
```

Comment:
[s3_spec.rb](https://github.com/Everlag/bosh/blob/e4d1d19ed80e8c50ff7bf39ede8165b32fc166bd/src/bosh-director/spec/functional/s3_spec.rb) has some tests for the read only case. Do you think it makes sense to add those in gcs_spec.rb now that this is supported?